Source/WebCore/PAL/pal/PALSwift/CryptoKitShim.swift

```diff
@@ -119,25 +131,322 @@ public class AesKw {
 
 }  // AesKw
 
+public enum HashFunction {
+    case sha256
+    case sha384
+    case sha512
+}
+
 public class Digest {
     public static func sha1(_ data: SpanConstUInt8) -> VectorUInt8 {
-        return digest(data, Insecure.SHA1.self)
+        return digest(data, t: Insecure.SHA1.self)
     }
     public static func sha256(_ data: SpanConstUInt8) -> VectorUInt8 {
-        return digest(data, SHA256.self)
+        return digest(data, t: SHA256.self)
     }
     public static func sha384(_ data: SpanConstUInt8) -> VectorUInt8 {
-        return digest(data, SHA384.self)
+        return digest(data, t: SHA384.self)
     }
     public static func sha512(_ data: SpanConstUInt8) -> VectorUInt8 {
-        return digest(data, SHA512.self)
+        return digest(data, t: SHA512.self)
     }
-    private static func digest<T: HashFunction>(_ data: SpanConstUInt8, _: T.Type)
-        -> VectorUInt8
+    fileprivate static func digest<T: CryptoKit.HashFunction>(_ data: SpanConstUInt8, _: T.Type)
+        -> T.Digest
     {
         var hasher = T()
         hasher.update(data: data)
-        return hasher.finalize().copyToVectorUInt8()
+        return hasher.finalize()
+    }
+
+    fileprivate static func digest<T: CryptoKit.HashFunction>(_ data: SpanConstUInt8, t: T.Type)
+        -> VectorUInt8
```

Comment:
It's probably worth having a better name than `digest(_:_:)` and `digest(_:t:)` where the latter makes a copy and the former doesn't. Maybe `digest(_:type:)` and `digestCopyingBytes(_:type:)`?

Reply:
They both are fileprivate and thus not exposed outbound so are `implementation` detail. Also, they both differ in their return type. Also, sidenote, every time we return the digest from Swift to Cpp, we make a copy.